### PR TITLE
Remove some obsolete/obscure internal define switches:

### PR DESCRIPTION
**Pull request by ekasper**
- FLAT_INC
- PKCS1_CHECK (the SSL_OP_PKCS1_CHECK options have been
  no-oped)
- PKCS_TESTVECT (debugging leftovers)
- SSL_AD_MISSING_SRP_USERNAME (unfinished feature)
- DTLS_AD_MISSING_HANDSHAKE_MESSAGE (unfinished feature)
- USE_OBJ_MAC (note this removes a define from the public header but
   very unlikely someone would be depending on it)
- STD_ERROR_HANDLE SSL_FORBID_ENULL

### CODE REVIEW

File: crypto/cryptlib.c [hunk 1]
```diff
@@ -167,22 +167,6 @@ void OPENSSL_showfatal(const char *fmta, ...)
     va_list ap;
     TCHAR buf[256];
     const TCHAR *fmt;
-# ifdef STD_ERROR_HANDLE        /* what a dirty trick! */
```
Comment (dot-asm) [inline crypto/cryptlib.c]: -1. "What a dirty trick" refers to ... Windows CE. I mean if not for Windows CE, then this #ifdef wouldn't exits. And I mean #ifdef itself[!], not code that it surrounds. In other words code it surrounds is normally used.

Comment (ekasper) [inline crypto/cryptlib.c]: Ah, this is some Win thing... can you suggest a comment that explains this ifdef?

Comment (dot-asm) [inline crypto/cryptlib.c]: The whole section is _WIN32, it shouldn't be surprise that it Win thing... As for commentary. "We first want to check if it's console application, in which case error message would be printed on standard error. However Windows CE doesn't have concept of console application. We could guard the check with #ifdef _WIN32_WCE, but #ifdef STD_ERROR_HANDLE does the trick just as good."

Comment (ekasper) [inline crypto/cryptlib.c]: Reverted the change, added a comment.

**Comment by dot-asm** [comment]
It looks like it needs re-base, as it's complaining about "conflicting files"...

**Comment by ekasper** [comment]
@dot-asm rebased

**Pull request closed without merging**